source/common/quic/platform/quiche_flags_impl.cc

```diff
@@ -17,6 +20,17 @@
 
 namespace {
 
+#define QUICHE_RELOADABLE_FLAG_OVERRIDE(flag_name, value)                                          \
+  {STRINGIFY(quic_reloadable_flag_##flag_name), value},
+constexpr std::pair<absl::string_view, bool> quiche_reloadable_flag_overrides[]{
```

Comment:
It would be nice if this were a map of some sort. (unfortunately, gtl::flat_map isn't available as part of Abseil... yet.)

Representing this as an array of pairs allows for overriding the same key multiple times, and in the implementation below, the first override wins.

(Also map semantics are nicer to work with than iterating over the entire array every time)

Reply:
The container needs to be constexpr compatible. Unfortunately most std containers are not, i.e. std::vector is not of literal type https://en.cppreference.com/w/cpp/named_req/LiteralType.

Reply:
Do you need this to be constexpr, or is it good enough to have static storage duration? (go/totw/110 isn't published externally, but among other things it suggests the pattern below)

absl::NoDestructor has been in OSS Abseil for a couple of months now: https://github.com/abseil/abseil-cpp/blob/master/absl/base/no_destructor.h

So you should be able to write something like...

```c++
const absl::NoDestructor<absl::flat_hash_map<absl::string_view, bool>>
  quiche_reloadable_flag_overrides([] {
    // some complex initialization logic
  }());
```

If you don't need complex initialization logic, you may be able to get away with just specifying an initializer list, e.g.

```c++
const absl::NoDestructor<absl::flat_hash_map<absl::string_view, bool>>
  quiche_reloadable_flag_overrides = {
    OVERRIDEN_RELOADABLE_FLAGS(QUICHE_RELOADABLE_FLAG_OVERRIDE)
};
```

(Even before we had NoDestructor, it was reasonable to just heap-allocate and "leak" the appropriate memory, e.g. `const auto* overrides = new flat_hash_map<...>(...);`)

Reply:
I can change the caller maybeOverride() not to be constexpr, so these don't have to be statically initialized. But absl::flat_hash_map doesn't seem to be trivially destructible though. And the guideline also suggests that global variables should be trivially destructible.

Reply:
Right, the idea is that absl::NoDestructor itself is trivially destructible, and it  doesn't really matter about the contained type if you never actually call its destructor.

That said, if this needs to be constexpr to satisfy maybeOverride, removing the constexpr qualifier is a big enough difference that the drawbacks of std::array that I mentioned above might be justified -- any cost might be paid during compile-time instead of runtime.

(That is: this is probably fine as-is.)

Reply:
absl::NoDestructor is not available in Envoy yet. That would require an abseil dep update. 

With remaining choice, I don't have a good sense about which one is better, assigning the desired flag values at compile time v.s. detecting duplicated flag names at startup. @RyanTheOptimist

Reply:
I don't think I feel strongly about I think steve's statement: "That is: this is probably fine as-is" works for me!

Reply:
Okay, I will keep it as is.